### PR TITLE
get active branches API

### PR DESCRIPTION
**Pull request by aryamanrishabh**
## Description
Implemented getBranchByTag API. It finds the active branches within a timeframe from a github repository.

## How did you test the changes?
Verified the correctness of the API response by invoking it through Postman and swagger locally.

## Testing Screenshots (Optional)
test-local

test-swagger

### CODE REVIEW

File: pkg/handler/branch.go [hunk 1]
```diff
@@ -35,12 +35,73 @@ func (h *branchHandler) GetActiveBranches(w http.ResponseWriter, r *http.Request
...
+		return nil, err
+	}
+
+	if resp.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("GitHub API returned status code: %d", resp.StatusCode())
+	}
+
+	var commitInfo = commit.Commit
+
+	return &commitInfo, nil
```
Comment (sarthakgoel1997) [inline pkg/handler/branch.go:110]: Just curious - since we are not using **commitInfo** in the parent function, can we skip returning it from here?

**Review by meiyu0714 — APPROVED** [review]

**Review by meiyu0714 — APPROVED** [review]

File: pkg/handler/branch.go (hunk 1 above)
Comment (aryamanrishabh) [reply pkg/handler/branch.go:110]: I am using this commit info in the parent function to find out if a branch has any commits within the specified period of time. Does that answer your question or was it something else?

Comment (sarthakgoel1997) [reply pkg/handler/branch.go:110]: Oh sorry, I meant the error part. Shouldn't we read the error and return it it is not nil?


https://github.com/nyuoss/git-service/pull/10/files#diff-36d91272e285a092a7c5a5ac717cac1ec6a31f3e39f59fa1ac7bd3546f6a5c5dR63

Comment (aryamanrishabh) [reply pkg/handler/branch.go:110]: Right, thank you for the review, I have updated the same

File: pkg/handler/branch.go [hunk 2]
```diff
@@ -60,7 +60,13 @@ func (h *branchHandler) GetActiveBranches(w http.ResponseWriter, r *http.Request
 
 	var activeBranches []string
 	for _, branch := range repoBranches {
-		commit, _ := getCommit(owner, repo, branch.Commit.SHA)
+		commit, err := getCommit(owner, repo, branch.Commit.SHA)
+
+		if err != nil {
+			http.Error(w, "Commit not found", http.StatusInternalServerError)
```
Comment (sarthakgoel1997) [inline pkg/handler/branch.go]: I think returning the same error from `getCommit` function would be more apt here. We can update "Commit not found" statement to **err.Error()** so we don't miss the original error message. What do you think?

Comment (aryamanrishabh) [reply pkg/handler/branch.go]: Yeah, that makes much more sense, thanks!
Sorry, I'm just struggling a bit with the syntax right now 😅

Comment (sarthakgoel1997) [reply pkg/handler/branch.go]: No worries, it is perfectly fine. Thank you for the changes!

**Review by sarthakgoel1997 — APPROVED** [review]

**Pull request merged**